### PR TITLE
Refactor: Add start and stop method to ResourceMonitor.

### PR DESCRIPTION
**Pull request by workingloong**
### What changes were proposed in this pull request?

Add start and stop method to ResourceMonitor.

### Why are the changes needed?

We can control the time to start and stop monitoring the resource.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

No.

### CODE REVIEW

File: dlrover/python/elastic_agent/monitor/resource.py [hunk 1]
```diff
@@ -92,32 +92,45 @@ def __init__(self):
...
-                    daemon=True,
-                )
-                thread.start()
-                if thread.is_alive():
-                    logger.info("Resource Monitor initialized successfully")
-            except Exception as e:
-                logger.error(
-                    f"Failed to start the monitor resource thread. Error: {e}"
-                )
+            return
```
Comment (Antlera) [inline dlrover/python/elastic_agent/monitor/resource.py:103]: Why we should return here when `MASTER_ADDR` & `AUTO_MONITOR` is `True`?

Comment (workingloong) [reply dlrover/python/elastic_agent/monitor/resource.py:103]: This is a bug and I have fixed it.

**Comment by Antlera** [comment]
LGTM

**Review by Antlera — APPROVED** [review]
LGTM

**Pull request merged**